extensions/typescript-language-features/src/tsServer/versionManager.ts

```diff
@@ -44,6 +51,19 @@ export class TypeScriptVersionManager extends Disposable {
 			});
 		}
 
+		this._register(vscode.workspace.onDidChangeWorkspaceTrustState((event: vscode.WorkspaceTrustStateChangeEvent) => {
+			if (this.useWorkspaceTsdkSetting) {
+				if (event.currentTrustState === vscode.WorkspaceTrustState.Trusted) {
+					const localVersion = this.versionProvider.localVersion;
+					if (localVersion) {
+						this.updateActiveVersion(localVersion);
+					}
```

Comment:
Do we need to call `this.updateActiveVersion(this.versionProvider.defaultVersion);` if the workspace is trusted but there is no local version?